compiler/rustc_const_eval/src/const_eval/mod.rs

```diff
@@ -89,6 +89,9 @@ pub(crate) fn try_destructure_mir_constant_for_user_output<'tcx>(
         }
         ty::Adt(def, _) => {
             let variant = ecx.read_discriminant(&op).ok()?;
+            if op.layout.for_variant(&ecx, variant).abi.is_uninhabited() {
+                return None;
+            }
```

Comment:
I don't think we should have such checks everywhere. That's too fragile. We should just remove the assertion: https://github.com/rust-lang/rust/pull/120367.